astropy/table/column.py

```diff
@@ -110,11 +55,15 @@ def _col_update_attrs_from(newcol, col, exclude_attrs=['name', 'parent_table']):
     if isinstance(newcol, BaseColumn):
         return
 
-    attrs = COLUMN_ATTRS - set(exclude_attrs)
+    if not hasattr(newcol, 'info'):
+        info_attr = '_info' if hasattr(col.__class__, 'info') else 'info'
```

Comment:
I'm a little confused why we check both the object and class for `info` here, and why it would help to write to `_info` is the object does not have `info` but the class does.